sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryAsyncClient.java

```diff
@@ -0,0 +1,186 @@
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<String> listRepositories(ListRepositoriesOptions options) {
+        if (options == null) {
+            throw logger.logExceptionAsError(new NullPointerException("options can't be null"));
+        }
+
+        Integer pageSize = options.getMaxPageSize();
+        return new PagedFlux<>(
+            () -> withContext(context -> listRepositoriesSinglePageAsync(pageSize, context)),
+            token -> withContext(context -> listRepositoriesNextSinglePageAsync(token, context)));
+    }
+
+    private PagedResponse<String> getPagedResponseWithContinuationToken(PagedResponse<String> listRepositoriesResponse) {
+        String continuationLink = null;
+        HttpHeaders headers = listRepositoriesResponse.getHeaders();
+
+        if (headers != null) {
+            String continuationLinkHeader = headers.getValue(CONTINUATIONLINKHEADERNAME);
+            if (!CoreUtils.isNullOrEmpty(continuationLinkHeader)) {
+                // TODO: ACR follows docker API spec which has continuation token in header.
+                // It currently returns it in the header without the url which they intend to fix
+                // in the upcoming releases.
+                Pattern linkPattern = Pattern.compile(LINKMATCHER);
+                Matcher matcher = linkPattern.matcher(continuationLinkHeader);
+                if (matcher.matches()) {
+                    if (matcher.groupCount() == 2) {
+                        String apiPath = matcher.group(2);
+                        continuationLink = apiPath;
+                    }
+                }
+            }
+        }
+
+        return new PagedResponseBase<String, String>(
+            listRepositoriesResponse.getRequest(),
+            listRepositoriesResponse.getStatusCode(),
+            listRepositoriesResponse.getHeaders(),
+            listRepositoriesResponse.getValue(),
+            continuationLink,
+            null
+        );
+    }
+
+    Mono<PagedResponse<String>> listRepositoriesSinglePageAsync(Integer pageSize, Context context) {
+        try {
+            Mono<PagedResponse<String>> pagedResponseMono = this.serviceClient.listRepositoriesSinglePageAsync(null, pageSize, context)
+                .map(res -> getPagedResponseWithContinuationToken(res));
+
+            return pagedResponseMono;
+
+        } catch (RuntimeException e) {
+            return monoError(logger, e);
+        }
+    }
+
+    Mono<PagedResponse<String>> listRepositoriesNextSinglePageAsync(String nextLink, Context context) {
+        if (nextLink == null) {
+            return Mono.error(logger.logExceptionAsWarning(new IllegalArgumentException("nextLink parameter cannot be null.")));
+        }
```

Comment:
This should never happen as `PagedFlux` doesn't continue on when the continuation token is null. Also, maybe return `Mono.empty()` as, even if we shouldn't be able to reach here, we know this is a terminal point in listing.

Reply:
Thanks. I will remove it :)

---
In reply to: [602630617](https://github.com/Azure/azure-sdk-for-java/pull/20139#discussion_r602630617) [](ancestors = 602630617)